src/GraphQL/ResolveFieldContext/ReadonlyResolveFieldContext.cs

```diff
@@ -49,20 +53,17 @@ private Fields GetSubFields()
         /// <inheritdoc/>
         public object Source => _executionNode.Source;
 
-        /// <inheritdoc/>
-        public string FieldName => _executionNode.Field.Name;
-
         /// <inheritdoc/>
         public Field FieldAst => _executionNode.Field;
 
         /// <inheritdoc/>
         public FieldType FieldDefinition => _executionNode.FieldDefinition;
 
         /// <inheritdoc/>
-        public IGraphType ReturnType => _executionNode.FieldDefinition.ResolvedType;
+        public IObjectGraphType ParentType => _executionNode.GetParentType(_executionContext.Schema);
 
         /// <inheritdoc/>
-        public IObjectGraphType ParentType => _executionNode.GetParentType(_executionContext.Schema);
+        public IResolveFieldContext Parent => _executionNode.Parent is RootExecutionNode || _executionNode.Parent == null ? null : _parent ??= new ReadonlyResolveFieldContext(_executionNode.Parent, _executionContext);
```

Comment:
Will this work for all situations, specifically array item nodes and children of array item nodes?

Reply:
Right. Fixed.